`x/tools@v0.1.6` fails to build on `go1.18` when running `make pre-commit -B`

<!-- If this is a regression, please open a regression report instead. -->

## Summary

When running `make pre-commit -B` on `go1.18`, `x/tools@v0.1.6` fails to build

```
go install golang.org/x/tools/cmd/goimports@v0.1.6
# golang.org/x/tools/internal/typeparams
../../../../pkg/mod/golang.org/x/tools@v0.1.6/internal/typeparams/typeparams_go118.go:78:6: sig.SetTypeParams undefined (type *types.Signature has no field or method SetTypeParams)
../../../../pkg/mod/golang.org/x/tools@v0.1.6/internal/typeparams/typeparams_go118.go:88:6: sig.SetRecvTypeParams undefined (type *types.Signature has no field or method SetRecvTypeParams)
../../../../pkg/mod/golang.org/x/tools@v0.1.6/internal/typeparams/typeparams_go118.go:98:15: iface.IsConstraint undefined (type *types.Interface has no field or method IsConstraint)
../../../../pkg/mod/golang.org/x/tools@v0.1.6/internal/typeparams/typeparams_go118.go:144:7: info.Inferred undefined (type *types.Info has no field or method Inferred)
../../../../pkg/mod/golang.org/x/tools@v0.1.6/internal/typeparams/typeparams_go118.go:144:42: undefined: types.Inferred
../../../../pkg/mod/golang.org/x/tools@v0.1.6/internal/typeparams/typeparams_go118.go:155:10: info.Inferred undefined (type *types.Info has no field or method Inferred)
../../../../pkg/mod/golang.org/x/tools@v0.1.6/internal/typeparams/typeparams_go118.go:158:14: info.Inferred undefined (type *types.Info has no field or method Inferred)
../../../../pkg/mod/golang.org/x/tools@v0.1.6/internal/typeparams/typeparams_go118.go:171:26: undefined: types.Environment
make: *** [Makefile:302: /home/rohan/go/bin/goimports] Error 2
```

See: https://github.com/golang/go/issues/48632

Should we use `x/tools@v0.1.7` in the `Makefile`?

---
<!-- Issue Author: Don't delete this message to encourage other users to support your issue! -->
**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

--- a/Makefile
+++ b/Makefile
@@ -299,7 +299,7 @@ $(GOPATH)/bin/openapi-gen:
 $(GOPATH)/bin/swagger:
 	go install github.com/go-swagger/go-swagger/cmd/swagger@v0.28.0
 $(GOPATH)/bin/goimports:
-	go install golang.org/x/tools/cmd/goimports@v0.1.6
+	go install golang.org/x/tools/cmd/goimports@v0.1.7
 
 pkg/apis/workflow/v1alpha1/generated.proto: $(GOPATH)/bin/go-to-protobuf $(PROTO_BINARIES) $(TYPES) $(GOPATH)/src/github.com/gogo/protobuf
 	# These files are generated on a v3/ folder by the tool. Link them to the root folder

